File: December - 06/KishoreMuruganantham_magic_square_odd.R
input <- file("stdin", "r")
line <- readLines(input, n = 1)
n_str <- sub(".*Enter n:\\s*", "", line)
n <- as.integer(n_str)

if (n %% 2 == 0) {
  cat("Magic square is only possible for odd values of n.\n")
} else {
  magic_const <- (n * (n^2 + 1)) / 2
  cat("Magic constant: ", magic_const, "\n", sep = "")
  
  grid <- matrix(0, nrow = n, ncol = n)
  r <- 1
  c <- (n + 1) / 2
  
  for (num in 1:(n^2)) {
    grid[r, c] <- num
    next_r <- r - 1
    next_c <- c + 1
    
    if (next_r < 1) next_r <- n
    if (next_c > n) next_c <- 1
    
    if (grid[next_r, next_c] != 0) {
      r <- r + 1
      if (r > n) r <- 1
    } else {
      r <- next_r
      c <- next_c
    }
  }
  
  for (i in 1:n) {
    cat(paste(grid[i, ], collapse = " "), "\n", sep = "")
  }
}

File: December - 06/KishoreMuruganantham_magic_square_odd.swift
import Foundation

if let line = readLine() {
    var nStr = line
    if let range = line.range(of: ":") {
        nStr = String(line[range.upperBound...]).trimmingCharacters(in: .whitespaces)
    }
    if let n = Int(nStr) {
        if n % 2 == 0 {
            print("Magic square is only possible for odd values of n.")
        } else {
            let magicConst = (n * (n * n + 1)) / 2
            print("Magic constant: \(magicConst)")
            var grid = Array(repeating: Array(repeating: 0, count: n), count: n)
            var r = 0
            var c = n / 2
            
            for num in 1...(n * n) {
                grid[r][c] = num
                var nr = r - 1
                var nc = c + 1
                if nr < 0 { nr = n - 1 }
                if nc >= n { nc = 0 }
                
                if grid[nr][nc] != 0 {
                    r = r + 1
                    if r >= n { r = 0 }
                } else {
                    r = nr
                    c = nc
                }
            }
            
            for i in 0..<n {
                print(grid[i].map{ String($0) }.joined(separator: " "))
            }
        }
    }
}
